geometry/create-and-edit-geometries/src/main/java/com/esri/samples/create_and_edit_geometries/CreateAndEditGeometriesController.java

```diff
@@ -0,0 +1,591 @@
+
+    // add example graphics to the map
+    createGraphicsMarkers();
+    createInitialGraphics();
+
+    // bind button availability to properties of the geometry editor
+    bindButtonsToGeometryEditor();
+
+    // allow user to immediately select and edit graphics
+    selectGraphic();
+  }
+
+  /**
+   * Binds button status to geometry editor properties.
+   */
+  private void bindButtonsToGeometryEditor() {
+    // point and multipoint disabled when freehand tool selected
+    pointButton.disableProperty().bind(geometryEditor.toolProperty().isEqualTo(freehandTool));
+    multipointButton.disableProperty().bind(geometryEditor.toolProperty().isEqualTo(freehandTool));
+
+    // only show the selected geometry type as enabled while editing
+    geometryEditor.startedProperty().addListener((ob, ov, nv) -> {
+      if (geometryEditor.isStarted()) {
+        pointButton.disableProperty().bind(new SimpleBooleanProperty(!geometryEditor.getGeometry().getGeometryType()
+          .equals(GeometryType.POINT)));
+        multipointButton.disableProperty().bind(new SimpleBooleanProperty(!geometryEditor.getGeometry().getGeometryType()
+          .equals(GeometryType.MULTIPOINT)));
+        polylineButton.disableProperty().bind(new SimpleBooleanProperty(!geometryEditor.getGeometry().getGeometryType()
+          .equals(GeometryType.POLYLINE)));
+        polygonButton.disableProperty().bind(new SimpleBooleanProperty(!geometryEditor.getGeometry().getGeometryType()
+          .equals(GeometryType.POLYGON)));
+      } else {
+        // when editor stopped, re-enable polygon & polyline, enable point/multipoint if vertex tool selected
+        pointButton.disableProperty().bind(geometryEditor.toolProperty().isEqualTo(freehandTool));
+        multipointButton.disableProperty().bind(geometryEditor.toolProperty().isEqualTo(freehandTool));
+        polylineButton.disableProperty().bind(new SimpleBooleanProperty(false));
+        polygonButton.disableProperty().bind(new SimpleBooleanProperty(false));
+      }
+    });
+
+    // bind the geometry editor tool to the tool chosen in the combo box
+    toolComboBox.valueProperty().bindBidirectional(geometryEditor.toolProperty());
+
+    // undo/redo enabled when the geometry editor can undo/redo
+    undoButton.disableProperty().bind(geometryEditor.canUndoProperty().not());
+    redoButton.disableProperty().bind(geometryEditor.canRedoProperty().not());
+
+    // delete selected element enabled only when an element is selected and deletable
+    ChangeListener<GeometryEditorElement> selectedElementChangedListener = (ob, ov, nv) -> {
+      if (nv == null) {
+        deleteSelectedElementButton.disableProperty().bind(new SimpleBooleanProperty(true));
+      } else {
+        deleteSelectedElementButton.disableProperty().bind(geometryEditor.getSelectedElement().canDeleteProperty().not());
+      }
+    };
+    geometryEditor.selectedElementProperty().addListener(selectedElementChangedListener);
```

Comment:
```suggestion
    geometryEditor.selectedElementProperty().addListener(((observable, oldValue, newValue) -> {
      if (newValue == null) {
        deleteSelectedElementButton.disableProperty().unbind();
        deleteSelectedElementButton.setDisable(true);
      } else {
        deleteSelectedElementButton.disableProperty().bind(geometryEditor.getSelectedElement().canDeleteProperty().not());
      }
    }));
```
Similarly, this is functionally the same but we can unbind and set the state. Also we can just set the listener directly as I don't think we will need to manually remove this listener (for example if we wanted to remove it later, we could use the reference in `.removeListener(...)`). I may be proved wrong here in a future review but I think it'll be garbage collected.